apps/wallet/src/composables/request-policies.composable.ts

```diff
@@ -45,3 +102,368 @@ export const useUserSpecifierSelectorItems = (): ComputedRef<SelectItem[]> => {
     return items;
   });
 };
+
+export type PopulatedUserSpecifier =
+  | {
+      kind: RequestPolicyRuleUserSpecifierEnum.Any;
+    }
+  | {
+      kind: RequestPolicyRuleUserSpecifierEnum.Id;
+      users: {
+        id: UUID;
+        name?: string;
+      }[];
+    }
+  | {
+      kind: RequestPolicyRuleUserSpecifierEnum.Group;
+      groups: {
+        id: UUID;
+        name?: string;
+      }[];
+    };
+
+export type PopulatedRule =
+  | {
+      kind: RequestPolicyRuleEnum.AutoApproved;
+    }
+  | {
+      kind: RequestPolicyRuleEnum.Quorum;
+      n: number;
+      approvers: PopulatedUserSpecifier;
+    }
+  | {
+      kind: RequestPolicyRuleEnum.QuorumPercentage;
+      n: number;
+      approvers: PopulatedUserSpecifier;
+    }
+  | {
+      kind: RequestPolicyRuleEnum.AllowListedByMetadata;
+      key: string;
+      value: string;
+    }
+  | {
+      kind: RequestPolicyRuleEnum.AllowListed;
+    }
+  | {
+      kind: RequestPolicyRuleEnum.AllOf;
+      rules: PopulatedRule[];
+    }
+  | {
+      kind: RequestPolicyRuleEnum.AnyOf;
+      rules: PopulatedRule[];
+    }
+  | {
+      kind: RequestPolicyRuleEnum.Not;
+      rule: PopulatedRule;
+    }
+  | {
+      kind: 'NamedRule';
+      name?: string;
+      id: UUID;
+      rule?: PopulatedRule;
+    };
+
+async function populateUserSpecifier(
```

Comment:
we should improve this in a separate PR to ensure we keep some caching on the get operations to avoid too many network requests

Reply:
Agreed, a cache in the app store with a short TTL would cover it I think.